src/v/pandaproxy/schema_registry/error.cc

```diff
@@ -71,6 +73,8 @@ struct error_category final : std::error_category {
             return reply_error_code::subject_version_soft_deleted; // 40406
         case error_code::subject_version_not_deleted:
             return reply_error_code::subject_version_not_deleted; // 40407
+        case error_code::subject_schema_invalid:
+            return reply_error_code::internal_server_error; // 500
```

Comment:
I'd tend to call this a 400 rather than a 500, are we doing this for compatibility with other implementations?

Reply:
I agree, it's completely the wrong status code. It's for compatibility.